sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images.asciidoc

```diff
@@ -1406,6 +1413,47 @@ Attempting to sample a standard sampled image with `sample_mipmap` or any other
 defined sampling function will result in undefined behaviour.
 ====
 
+==== `gather_image`
+
+[NOTE]
+=====
+`gather_image` is only currently supported in the Cuda backend.
+=====
+
+`DataT` is a vector type with four components. DataT can only be `sycl::float4`
+, `sycl::int4`, or `sycl::uint4`. `ImageHandle` must be a two dimensional image
```

Comment:
What mechanism enforces the "must"? The document should say what happens when these conditions are violated - options are "implementation defined", "unspecified", "undefined behavior", "error" (stating which error).

Reply:
At the moment it just doesn't compile because I've used std::enable_if to disallow any other types. This I think follows the pattern of other apis that implement sycl spec APIs, so I think that this behaviour is already consistent with what is expected in the SYCL spec. I might be wrong however.

Reply:
The sycl spec for math functions that only support certain types give this information only as far as I can see:

e.g.
```

Constraints: Available only if all of the following conditions are met:

    NonScalar is marray, vec, or the __swizzle__ type; and

    The element type is float, double, or half.


```

The corresponding implementations of these math functions use `enable_if` to disallow other types. So then the user gets a compilation error stating that the overload they are attempting isn't defined.

Reply:
Yeah I think we'll need to update that at some point, it's OK for now.